File: WebApi/DAL/Models/CartItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WebApi.DAL.Models
{
    public class CartItem
    {
        public int Id { get; set; }

        [ForeignKey("ProductId")]
        public virtual Product Product { get; set; }
        public int ProductId { get; set; }

        public string productName { get; set; }
        public int qty { get; set; }
        public float Price { get; set; }
        public int DiscountPercentage { get; set; }


    }
}

File: Angular Front End/src/app/models/cart-item.ts
import { Product } from './product';

export class CartItem {
  id: number;
  productId: number;
  productName: string;
  qty: number;
  price: number;
  discountPercentage:number

  constructor(id: number, product: Product, qty = 1) {
    this.id = id;
    this.productId = product.id;
    this.productName = product.name;
    this.price = product.price;
    this.qty = qty;
    this.discountPercentage =product.discountPercentage
  }
}
